Includes/Core/Field/VectorField.hpp

```diff
@@ -0,0 +1,78 @@
+// This code is based on Jet framework.
+// Copyright (c) 2018 Doyub Kim
+// CubbyFlow is voxel-based fluid simulation engine for computer games.
+// Copyright (c) 2020 CubbyFlow Team
+// Core Part: Chris Ohk, Junwoo Hwang, Jihong Sin, Seungwoo Yoo
+// AI Part: Dongheon Cho, Minseo Kim
+// We are making my contributions/submissions to this project solely in our
+// personal capacity and are not conveying any rights to any intellectual
+// property of any third parties.
+
+#ifndef CUBBYFLOW_VECTOR_FIELD_HPP
+#define CUBBYFLOW_VECTOR_FIELD_HPP
+
+#include <Core/Field/Field.hpp>
+#include <Core/Matrix/Matrix.hpp>
+
+#include <functional>
+#include <memory>
+
+namespace CubbyFlow
+{
+template <size_t N>
+struct GetCurl
+{
+    // Do nothing
+};
+
+template <>
+struct GetCurl<2>
+{
+    using Type = double;
```

Comment:
Is it possible to define curl for 2D? I thought it would be valid for 3D since we can define the cross product of two vectors in 3D space.

Reply:
Yes. It's possible.
https://www.khanacademy.org/math/multivariable-calculus/greens-theorem-and-stokes-theorem/formal-definitions-of-divergence-and-curl/a/defining-curl